File: leetcode-swift/lc_007.swift
//
// LeetCode
// 7. Reverse Integer
// https://leetcode.com/problems/reverse-interger
//


class Solution {
    func reverse(_ x: Int) -> Int {
        var num = x
        var digit: Int
        var result = 0
        
        while num != 0 {
            digit = num % 10
            
            if num > 0 && Int32(result) > (Int32.max - Int32(digit)) / 10 {
                return 0
            }
            
            if num < 0 && Int32(result) < (Int32.min - Int32(digit)) / 10 {
                return 0
            }
            
            result = result * 10 + digit
            num /= 10
        }
        
        return result
    }
}

File: leetcode-go/lc_007.go
//
// LeetCode
// 7. Reverse Integer
// https://leetcode.com/problems/reverse-interger
//


func reverse(x int) int {
    var digit int
    var result int

    for x != 0 {
        digit = x % 10

        if x > 0 && int32(result) > (math.MaxInt32 - int32(digit)) / 10 {
            return 0
        }

        if x < 0 && int32(result) < (math.MinInt32 - int32(digit)) / 10 {
            return 0
        }

        result = result * 10 + digit
        x /= 10
    }

    return result
}
